docs/beam_handling.rst

```diff
@@ -53,19 +53,122 @@ Multi-beam cubes
+as with regular `~spectral_cube.spectral_cube.SpectralCube` s. For example, to
+convert from Jy / beam to K::
 
-You can identify channels with bad beams (i.e., beams that differ from a reference beam,
-which by default is the median beam) using
+    >>> vrsc_cube_K = vrsc_cube.to(u.K)  # doctest: +SKIP
+
+``spectral-cube`` will use a different beam and frequency for each plane.
+
+Handling variations in the beams is often a source of difficulty. Some spectral
+operations (e.g., moment maps) require a common resolution. However, a few channels
+may have large discprepancies in the beam compared to most of the others (e.g., if
+more data in a particular channel was bad and had to be flagged).
+`~spectral_cube.spectral_cube.VaryingResolutionSpectralCube` deals with these problems in two ways.
+
+Finding and convolving to a common resolution
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When applying spectral operations on a
+`~spectral_cube.spectral_cube.VaryingResolutionSpectralCube`, it is often
+necessary to first convolve the data to have a common beam. To retain the
+maximum amount of spatial information, the data can be convolved to the
+smallest common beam that all beams in
+`~spectral_cube.spectral_cube.VaryingResolutionSpectralCube.beam` can be
+deconvolved from.
+
+Finding the smallest common beam and convolution operations are described
+in detail in :doc:`smoothing`.
+
+
+Small beam variations: Do I need to convolve first?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Formally, operations on multiple spectral channels require the data to have a common
+resolution. In practice, data cubes with a high-spectral resolution will have the beam
+vary by a very small factor between adjacent channels. It is then a safe approximation
+to ignore small changes in the beam to avoid having to convolve the entire data cube to
+a common beam (which might be a slow and expensive operation for large cubes).
+
+`~spectral_cube.spectral_cube.VaryingResolutionSpectralCube` allows for small variations
+in the beam by setting
+:meth:`~spectral_cube.spectral_cube.VaryingResolutionSpectralCube.beam_threshold`. The
+`beam_threshold` is the largest allowed fractional change relative to the smallest common beam that will be allowed in spectral operations. For example, the default 0.01 allows up
+to a 1% variation from the smallest common beam. If the beams vary more than this, a
+`ValueError` will be raised prior to the spectral operation. Otherwise, variations between
+the beams will be ignored, a warning will be printed describing this approximation, and
+the spectral operation will be computed.
+
+The beam threshold can be changed from the default 0.01 with::
+
+    >>> vrsc_cube.beam_threshold = 0.02   # doctest: +SKIP
+
+.. note::
+
+    Setting `vrsc_cube.beam_threshold = 1.0` will allow all beam variations without raising an error. Large values for the beam threshold should not be used for scientific results.
```

Comment:
I think this is not true. You can have more than 100% variation in beam size, then this will still fail.  I've had this happen when there was a 7m beam in 1 channel and real 12m beams in the others

Reply:
Fixed. Noted that 1.0 and larger allow for large beam variations (not "all").